tests/Tokenizer/TokensTest.php

```diff
@@ -873,44 +876,100 @@ public function testEnsureWhitespaceAtIndex($expected, $input, $index, $offset,
         $tokens = Tokens::fromCode($input);
         $tokens->ensureWhitespaceAtIndex($index, $offset, $whiteSpace);
 
-        $this->assertSame($expected, $tokens->generateCode());
+        $this->assertTokens(Tokens::fromCode($expected), $tokens);
     }
 
     public function provideEnsureWhitespaceAtIndexCases()
     {
         return [
             [
+                '<?php $a. $b;',
+                '<?php $a.$b;',
+                2,
+                1,
+                ' ',
+            ],
+            [
+                '<?php $a .$b;',
+                '<?php $a.$b;',
+                2,
+                0,
+                ' ',
+            ],
+            [
+                '<?php
+//
+ echo $a;',
+                '<?php
+//
+echo $a;',
+                2,
+                1,
+                "\n ",
+            ],
+        ];
+    }
+
+    /**
+     * @param string $expected   valid PHP code
+     * @param string $input      valid PHP code
+     * @param int    $index      token index
+     * @param int    $offset
+     * @param string $whiteSpace white space
+     *
+     * @dataProvider provideEnsureWhitespaceAtIndexCasesWithOpenTag
+     */
+    public function testEnsureWhitespaceAtIndexWithOpenTag($expected, $input, $index, $offset, $whiteSpace)
+    {
+        if (ini_get('short_open_tag')) {
+            $this->markTestSkipped('The short_open_tag option is required to be disabled.');
```

Comment:
Please add a note about why this is required (either in the skip message or as a comment).